app/models/category.rb

```diff
@@ -0,0 +1,6 @@
+class Category < ApplicationRecord
+  validates :name, presence: true
+
+  has_and_belongs_to_many :recipes
+  accepts_nested_attributes_for :recipes
+end
```

Comment:
This would be a useful helper method for getting categories by names.
```suggestion

  def self.from_names(names)
    names.map do |name|
      find_or_create_by(name: name)
    end
  end
end
```
  The `self` here means that it is a class method rather than an instance method. Meaning you call it like:

```
categories = Category.from_names(%w[Main Side Appetizer NewThing])
```